src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java

```diff
@@ -473,6 +479,58 @@ void hanldeTopicCreateOperation(final O topicOp, final Map<String, KafkaFuture<V
     topicOp.finishRequest();
   }
 
+  void readTopics(final List<O> topicOps) {
+    // TODO Create cache for Kafka Consumer
+    val nodeAddr = topicOps.get(0).nodeAddr();
+    try {
+      concurrencyThrottle.acquire();
+      for (var i = 0; i < topicOps.size(); i++) {
+        val topicOp = (PathOperation) topicOps.get(i);
+        val topicItem = topicOp.item();
+        val topicName = topicItem.name().replaceAll("/", "");
+        try {
+          val consumerConfig = createConsumerConfig(nodeAddr);
+          consumerConfig.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, Integer.MAX_VALUE);
+          topicOp.startRequest();
+          val kafkaConsumer = new KafkaConsumer<String, byte[]>(consumerConfig);
+          kafkaConsumer.subscribe(Arrays.asList(topicName));
+          topicOp.finishRequest();
+          topicOp.startResponse();
+          var sizeOfReadData = 0;
+          var records = kafkaConsumer.poll(recordOpTimeout);
+          for (var record : records) {
+            sizeOfReadData += record.value().length;
+          }
+          while (!records.isEmpty()) {
+            records = kafkaConsumer.poll(recordOpTimeout);
+            for (var record : records) {
+              sizeOfReadData += record.value().length;
+            }
+          }
+          topicOp.finishResponse();
+          topicOp.countBytesDone(sizeOfReadData);
+          kafkaConsumer.close();
```

Comment:
and if earlier there will be an exception?

Reply:
Ok, I add finally block here